v2/cmd/controller/app/setup.go

```diff
@@ -44,6 +50,40 @@ import (
 	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
 )
 
+func SetupPreUpgradeCheck(ctx context.Context) error {
+	cfg := clientconfig.GetConfigOrDie()
```

Comment:
IMO use the `GetConfig` flavor since we're returning an error here. `OrDie` is really only appropriate in a top level caller that doesn't return an error.

Since you _are_ returning an error here you can just return the error from `GetConfig` too.